stdlib/bytes.mli

```diff
@@ -319,6 +319,21 @@ val equal: t -> t -> bool
 (** The equality function for byte sequences.
     @since 4.03.0 *)
 
+val split_on_char: char -> bytes -> bytes list
+(** [split_on_char sep s] returns the list of all (possibly empty)
+    substrings of [s] that are delimited by the [sep] character.
```

Comment:
```suggestion
    subarrays of [s] that are delimited by the [sep] character.
```
Since `bytes` are "byte arrays" I think we should avoid the "substring" terminology.

Reply:
It felt strange to me writing subarrays - we describe them as "Byte sequence operations", so I went with subsequences if that's OK?